users/init.sls

```diff
@@ -59,9 +74,12 @@ user_{{ name }}_sshdir:
 
   {% for k in u.sshpubkeys|default([]) %}
 user_{{ name }}_ssh_auth_{{ k.key[-20:] }}:
-  ssh_auth:
-    - {{ k.ensure|default('present') }}
```

Comment:
Ah OK, now I understand what you did here.
you can ensure that a key gets deleted with "ensure: absent" nice!